src/linter/linter.ts

```diff
@@ -1,21 +1,87 @@
 import {graphFromObject} from "@ui5/project/graph";
-import {createReader, createWorkspace, createReaderCollection} from "@ui5/fs/resourceFactory";
+import {createReader, createWorkspace, createReaderCollection, createFilterReader} from "@ui5/fs/resourceFactory";
 import {FilePath, LinterOptions, LintResult} from "./LinterContext.js";
 import lintWorkspace from "./lintWorkspace.js";
 import {taskStart} from "../utils/perf.js";
 import path from "node:path";
 import posixPath from "node:path/posix";
 import {stat} from "node:fs/promises";
 import {ProjectGraph} from "@ui5/project";
-import {AbstractReader} from "@ui5/fs";
+import type {AbstractReader, Resource} from "@ui5/fs";
+import ConfigManager from "../utils/ConfigManager.js";
+import {minimatch} from "minimatch";
 
 async function lint(
 	resourceReader: AbstractReader, options: LinterOptions
 ): Promise<LintResult[]> {
 	const lintEnd = taskStart("Linting");
+	let {ignorePattern} = options;
+
+	let fsBasePath = "";
+	let fsBasePathTest = "";
+	let virBasePath = "/resources/";
+	let virBasePathTest = "/test-resources/";
+	let projectRootDir = options.rootDir;
+	try {
+		const graph = await getProjectGraph(options.rootDir);
+		const project = graph.getRoot();
+		projectRootDir = project.getRootPath();
+		fsBasePath = project.getSourcePath();
+		fsBasePathTest = path.join(projectRootDir, project._testPath ?? "test");
+
+		if (!project._isSourceNamespaced) {
+			// Ensure the virtual filesystem includes the project namespace to allow relative imports
+			// of framework resources from the project
+			virBasePath += project.getNamespace() + "/";
+			virBasePathTest += project.getNamespace() + "/";
+		}
+	} catch {
+		// Project is not resolved i.e. in tests
+	}
+
+	const relFsBasePath = path.relative(options.rootDir, fsBasePath);
+	const relFsBasePathTest = fsBasePathTest ? path.relative(options.rootDir, fsBasePathTest) : undefined;
+
+	const configMngr = new ConfigManager(projectRootDir);
+	const config = await configMngr.getConfiguration();
+	ignorePattern = [
+		...(config.ignores ?? []),
+		...(ignorePattern ?? []), // CLI patterns go after config patterns
+	].filter(($) => $);
+
+	// Patterns must be only relative (to project's root),
+	// otherwise throw an error
+	ignorePattern.forEach((pattern) => {
+		let notNegatedPattern = pattern;
+		if (pattern.startsWith("!")) {
+			notNegatedPattern = pattern.slice(1);
+		}
+
+		if (/\\/g.test(pattern)) {
```

Comment:
Not sure whether I should open this can of worms, because I think we have similar code already elsewhere. But it just occurred to me that on UNIX, a file- or directory name can contain backslashes, e.g. `My\ File.js` is a valid file name on macOS.

So I'm not sure whether we should have this check. Maybe it's enough that paths with non-POSIX separators simply won't match anything?

Reply:
As we have agreed, removing this validation, until we find a reasonable solution